source/middleware/AdminCheckMiddleware.ts

```diff
@@ -0,0 +1,41 @@
+import { type Subsection, type Section, type User, type Listing } from '../models/index.js';
+
+export function isLoggedOn (localuser: undefined | null | User): localuser is User {
+	return localuser !== null && localuser !== undefined;
+}
+
+export function doesUserExist (user: undefined | null | User): user is User { // Code readibility and context wise this may be used but is identical to isLoggedOn
+	return user !== null && user !== undefined;
+}
```

Comment:
While the function is the same currently, it may be the case that functionality splits down the line so i'm all for this.